knimin/lib/parse.py

```diff
@@ -94,3 +95,84 @@ def parse_qpcr_object(contents):
         result[r, c] = value
 
     return result
+
+
+def _find_section(lines, label):
+    """Find a section, eg [EXCEPTIONS]
+
+    Parameters
+    ----------
+    lines : list of str
+        The data to search over
+    label : str
+        The section header to search for
+
+    Returns
+    -------
+    int
+        The index of the header
+    int
+        The index of the first empty line following the section, or the index
+        position of the end of the file.
+
+    Raises
+    ------
+    ValueError
+        If the section is not found
+    """
+    start = -1
```

Comment:
I feel that this calculation is overly complicated. Maybe possible to interate the `lines` only once. For example:
```
start = -1
end = len(lines)
started = False
for idx, line in enumerate(lines):
    if line.startswith(label):
        start = idx
        started = True
    elif started and not line:
        end = idx
        break
if start == -1:
    raise ValueError('%s section appears to be missing' % label)
```